### PR TITLE
Add unit test for spawning simple_animals

### PR DESCRIPTION
**Pull request by Gamer025**
## About The Pull Request
Simple unit test that allocates all mobs a gold slime could spawn

## Why It's Good For The Game
Unit test that ensures that the most common simple_animals can be spawned and deleted without runtime

### CODE REVIEW

File: code/modules/unit_tests/spawn_mobs.dm [hunk 1]
```diff
@@ -0,0 +1,8 @@
+///Unit test that spawns all mobs that can be spawned by golden slimes
+/datum/unit_test/spawn_mobs
+
+/datum/unit_test/spawn_mobs/Run()
+	for(var/T in typesof(/mob/living/simple_animal))
```
Comment (Jared-Fogle) [inline code/modules/unit_tests/spawn_mobs.dm:5]: Please change this to _animal, I dislike single letter variable names like this (especially when they're used for more than just a type cast).

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Just one pedantic style concern.

I fully expect this to be one of those sporadically failing tests, but I think we can cross that bridge when we get there.

File: code/modules/unit_tests/spawn_mobs.dm (hunk 1 above)
Comment (Gamer025) [reply code/modules/unit_tests/spawn_mobs.dm]: Yeah I must admit I kinda copied the for from mob.dm and forgot to change the var name, fixed now

**Comment by Gamer025** [comment]
> Just one pedantic style concern.
> 
> I fully expect this to be one of those sporadically failing tests, but I think we can cross that bridge when we get there.

Well I probably ran the test at least 5 times and it never runtimed, so lets hope not.

**Review by Mothblocks — APPROVED** [review]

**Comment by Jared-Fogle** [comment]
That's because the RNG is seeded. Eventually, someone will change the order RNG is called in and the test will maybe fail.

**Pull request merged**